### ISSUE TITLE
HTTP Connection Pool Possibility (Question) ?

### ISSUE THREAD

**Issue by karlandoh** [ccxt/ccxt#3122]
CCXT Version: 1.14.180 (latest).
Language: Python

Hope things are great. Just a quick question. I have a lot of CCXT requests firing away at once, and I constantly get timeout errors. Now I've referred to the manual and I know how to raise an exception in order to 'capture' the timeout error and send again. However, after searching throughout the issues database, it seems like no one has mentioned if it was possible to fetch many CCXT requests at once through an HTTPConnection pool. Is this possible? This would help not only remove my timeout errors but also take care of my **HTTP NewConnection Errors** as well.

Apologizes for the naivety! Thanks.

**Comment by kroitor** [comment]
Hi! Please pay attention to sections on rate limiting:

- https://github.com/ccxt/ccxt/wiki/Manual#rate-limit
- https://github.com/ccxt/ccxt/wiki/Manual#authentication
- https://github.com/ccxt/ccxt/search?q=ratelimit&unscoped_q=ratelimit&type=Issues
- https://github.com/ccxt/ccxt/search?q=rate+limiting&unscoped_q=ratelimit&type=Issues
- https://github.com/ccxt/ccxt/tree/master/examples/py (see examples on rate limiting here)

Basically, you're observing rate-limiting and you should throttle your requests accordingly. You should either use the built in limiter as shows in the examples or use your own. Hope this answers your question.

**Comment by karlandoh** [comment]
Hey. I forgot to add that I'm throttling many requests from about 10K proxies, selected at random, so I shouldn't be coming across that issue (each proxy is being used about once a minute from my logging file). However, I am getting backed up. I've strategically allocated my memory so that's not the issue. I'm just not sure where it's being backed up...

Thanks.
